### PR TITLE
Refactor localization

### PR DESCRIPTION
**Pull request by Otisz**
Hello!

This is my first ever LUA and addon development. I wanted to get experience before I make more changes.

This PR brings a _new_ approach to handle localization.

I was highly inspired by [ALL THE THINGS](https://www.curseforge.com/wow/addons/all-the-things) and other tutorials.

## What's new

- Every language has it's own file. Simplier to read.
- Every non-english file contains the missing localization as comment.
- The order of the keys is the same as in the `enUS.lua`. (It was completely unnecessary, but it made the comparing easier.)

## Known bugs

- ~~Currently only the english version works.
This is because how LUA tables works. In the `enUS.lua`  the initial set, and getting the values in other files are okay, but when it needs to set an other language, it fails because some keys contains space ` `.~~

## Upcoming changes

- ~~As mentioned above, I need to replace the keys which contains space.~~ (done)
- ~~To keep consistency, every key needs to be replaced.~~ (done)
- ~~Find and translate hardcoded texts (Should I use Google Translate for those?)~~ (postpone)

## Notes

- `JOL_NOTES = "Get Clotthing Powder"`  and `JOL_POWEDER = "Get Jols Notes"` are swapped.
I did not check the usage.

- (probably) unused:
  - BUY_ITEM
  - COMPLETE_QUEST
  - SETTINGS
  - CHOOSE_QUESTS
  - BLOODLUST
  - USE_PORTAL_TO_ZULDAZAR
  - IMMUNE_DURING_RP
  - REDUCED_DAMAGE_INFO_2
  - AUTO_SHARE_QUESTS
  - SHOW_BANNER_COOLDOWNS
  - BANNER_WINDOW_SCALE
  - WAR_CAMPAIGN

### CODE REVIEW

File: APR-Core/QuestHandler.lua [hunk 1]
```diff
@@ -925,7 +928,7 @@ local function APR_PrintQStep()
 		if (steps["Bloodlust"] and APR.ZoneTransfer == 0) then
 			if (APR1[APR.Realm][APR.Name]["Settings"]["ShowQList"] == 1) then
 				LineNr = LineNr + 1
-				APR.QuestList.QuestFrames["FS"..LineNr]:SetText("** "..APR_Locals["Bloodlust"].." **")
+				APR.QuestList.QuestFrames["FS"..LineNr]:SetText("** "..APR_LLocals["Bloodlust"].." **")
```
Comment (Aldric-Ducreux) [inline APR-Core/QuestHandler.lua:931]: Typo : shoulb be `APR.QuestList.QuestFrames["FS"..LineNr]:SetText("** "..L["BLOODLUST "].." **")`

**Comment by Aldric-Ducreux** [comment]
Hello @Otisz ,

I'm not a dev of the project but I've been browsing your code to see what you've done. 

So, if I understand correctly what you did (copying the principle of All The Things).
In your code `local L = app.L;` is for "Localizations"(if I don't say anything stupid), I prefer to have variable names that are readable and understandable from everywhere in the code.
So, I think renaming L to Locals/Localizations/... might be better. 

Also, from what I can see in the code, the translation is not present in all the `print`. 
It should be added and if necessary also add in the i18n locale files, if you want to have a full translation.

**Exemple:** 
`Core.lua line 442`
```lua
-- Chat commands, such as /apr reset, /apr skip, /apr skipcamp
local function APR_SlashCmd(APR_index)
	if (APR_index == "reset") then --Command for making the quest rescan on completion and reset, including previously skipped steps
		print("APR: Resetting Zone.")
		APR1[APR.Realm][APR.Name][APR.ActiveMap] = 1
		APR.BookingList["UpdateQuest"] = 1
		APR.BookingList["PrintQStep"] = 1
	elseif (APR_index == "skip") then
		print("APR: Skipping Quest Step.") -- Command for skipping the current quest step
		APR1[APR.Realm][APR.Name][APR.ActiveMap] = APR1[APR.Realm][APR.Name][APR.ActiveMap] + 1
		APR.BookingList["UpdateQuest"] = 1
		APR.BookingList["PrintQStep"] = 1
	elseif (APR_index == "skipcamp") then
		print("APR: Skipping Camp Step.") -- Command for skipping "camp" step
		APR1[APR.Realm][APR.Name][APR.ActiveMap] = APR1[APR.Realm][APR.Name][APR.ActiveMap] + 14
		APR.BookingList["UpdateQuest"] = 1
		APR.BookingList["PrintQStep"] = 1
	else
		APR.SettingsOpen = 1
		APR.OptionsFrame.MainFrame:Show()
		APR.RemoveIcons()
		APR.BookingList["OpenedSettings"] = 1
	end
end
```

PS: I can do the FR translation if needed

**Comment by Otisz** [comment]
Hello @Aldric-Ducreux,

Well, the letter `L` is not most descriptive for a variable name, but as I looked into other addons, turned out it is a naming convention.

I think we can both agree that BigWig is a popular and big addon. [They have a packager](https://github.com/BigWigsMods/packager), that everybody can use, and it downloads the locales from CurseForge and place them into the code. It uses `L` as the name of the variable.

---

As you can see the 3rd point in the Upcoming changes, I postponed it.
There is 2-3 times more untranslated text in the project. It would take a lot of time. They were never translated anyway, so it can wait a little more.

**Comment by Aldric-Ducreux** [comment]
Thanks for the clarification @Otisz 
So I no longer have a problem with using `L` :)

And Sorry, I didn't see that the upcoming changes. If you need help to find out what is hardcoded, I can help you. 

---

For the translations, it's not a really good idea to use Google translate or DeepL. I've just done the test with the Fench file and it's not very good. 
the thing is this kind of translator doesn't have the knowledge of the technical words of WoW/MMORPG.

It's better to add the text in the translation files and then use English as default if there is no translation available.

**Review by Neogeekmo — APPROVED** [review]

**Comment by Rycia** [comment]
We've talked it over and we're going to merge your request.

This will help us performance-wise by not having to run one file as opposed to one of multiple files, henceforth better ram usage. It'll create a better ground for future expansion, especially in localization.

**Pull request merged**